File: frontend/src/api/audit.ts
import { apiClient } from "./client"
import type { AuditEntry } from "@/types"

/** The current user's recent security/account audit events (newest first). */
export async function listAudit(limit = 25): Promise<AuditEntry[]> {
  const { data } = await apiClient.get<AuditEntry[]>("/api/audit", { params: { limit } })
  return data
}

File: backend/app/routers/audit.py
"""Audit log router — the user's own security/account event trail."""
from __future__ import annotations

from fastapi import APIRouter, Depends, Query
from sqlalchemy import select
from sqlalchemy.ext.asyncio import AsyncSession

from app.database import get_db
from app.dependencies.auth import get_current_user
from app.models.audit_log import AuditLog
from app.models.user import User
from app.schemas.audit import AuditLogOut

router = APIRouter(prefix="/api/audit", tags=["audit"])


@router.get("", response_model=list[AuditLogOut])
async def list_audit(
    limit: int = Query(default=50, le=200),
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
) -> list[AuditLog]:
    """Return the current user's recent audit events, newest first."""
    result = await db.execute(
        select(AuditLog)
        .where(AuditLog.user_id == current_user.id)
        .order_by(AuditLog.created_at.desc())
        .limit(limit)
    )
    return list(result.scalars().all())
